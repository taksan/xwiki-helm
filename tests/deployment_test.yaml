suite: Deployment

templates:
  - deployment.yaml

tests:
  - it: should use custom image
    set:
      image.tag: latest
      mysql.enabled: false
      postgresql.enabled: false
      image.name: default
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: default:latest
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: default-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -latest

  - it: should use mysql lts image
    set:
      image.tag: latest
      mysql.enabled: true
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-mysql-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-mysql-tomcat
      - equal:
          path: spec.template.spec.containers[0].env[1].valueFrom.secretKeyRef.key
          value: mysql-password
  - it: should use postgres lts image
    set:
      image.tag: latest
      postgresql.enabled: true
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-postgres-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-postgresql-tomcat
      - equal:
          path: spec.template.spec.containers[0].env[1].valueFrom.secretKeyRef.key
          value: postgresql-password
  - it: should use mysql lts image when mariadb is specified as it is compatible with mysql
    set:
      image.tag: latest
      mariadb.enabled: true
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-mysql-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-mariadb-tomcat
      - equal:
          path: spec.template.spec.containers[0].env[1].valueFrom.secretKeyRef.key
          value: mariadb-password
  - it: should properly set value
    set:
      image.pullPolicy: IfNotPresent
      service.internalPort: 8080
      mysql.enabled: true
    asserts:
      - equal:
          path: metadata.name
          value: RELEASE-NAME-xwiki
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
      - equal:
          path: spec.template.spec.containers[0].ports[0].containerPort
          value: 8080
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].name
          value: xwiki-solr-data
      - equal:
          path: spec.template.spec.volumes[0].persistentVolumeClaim.claimName
          value: xwiki-solr
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[1].name
          value: xwiki-solr-data
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[2].name
          value: xwiki-tomcat-conf
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[2].mountPath
          value: /usr/local/tomcat/conf/web.xml
      - matchSnapshot:
          path: spec.template.spec.containers[0].volumeMounts
      - matchSnapshot:
          path: spec.template.spec.volumes
