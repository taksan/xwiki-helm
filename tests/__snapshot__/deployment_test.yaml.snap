should properly set value:
  1: |
    - mountPath: /usr/local/xwiki
      name: xwiki-data
    - mountPath: /opt/solr/server/solr
      name: xwiki-solr-data
    - mountPath: /usr/local/tomcat/conf/web.xml
      name: xwiki-tomcat-conf
      subPath: web.xml
    - mountPath: /usr/local/xwiki/data/xwiki.cfg
      name: xwiki-cfg
      subPath: xwiki.cfg
    - mountPath: /usr/local/xwiki/data/xwiki.properties
      name: xwiki-properties
      subPath: xwiki.properties
  2: |
    - name: xwiki-solr-data
      persistentVolumeClaim:
        claimName: xwiki-solr
    - name: xwiki-data
      persistentVolumeClaim:
        claimName: xwiki
    - configMap:
        items:
        - key: web.xml
          path: web.xml
        name: xwiki-tomcat-conf
      name: xwiki-tomcat-conf
    - configMap:
        items:
        - key: xwiki.cfg
          path: xwiki.cfg
        name: xwiki-cfg
      name: xwiki-cfg
    - configMap:
        items:
        - key: xwiki.properties
          path: xwiki.properties
        name: xwiki-properties
      name: xwiki-properties
should use custom image:
  1: |
    - name: JAVA_OPTS
      value: -Xmx2048m
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: DB_PASSWORD
          name: RELEASE-NAME
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use mysql lts image:
  1: |
    - name: JAVA_OPTS
      value: -Xmx2048m
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mysql-password
          name: RELEASE-NAME-mysql
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use mysql lts image when mariadb is specified as it is compatible with mysql:
  1: |
    - name: JAVA_OPTS
      value: -Xmx2048m
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mariadb-password
          name: RELEASE-NAME-mariadb
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use postgres lts image:
  1: |
    - name: JAVA_OPTS
      value: -Xmx2048m
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: postgresql-password
          name: RELEASE-NAME-postgresql
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
